pandas/core/frame.py

```diff
@@ -3353,38 +3353,23 @@ def assign(self, **kwargs):
 
         Examples
         --------
-        >>> df = pd.DataFrame({'A': range(1, 11), 'B': np.random.randn(10)})
+        >>> df = pd.DataFrame({'A': range(1, 3),
+        ...                    'B': np.arange(-1.0, 2.0, 1.5)})
 
         Where the value is a callable, evaluated on `df`:
 
         >>> df.assign(ln_A = lambda x: np.log(x.A))
-            A         B      ln_A
-        0   1  0.426905  0.000000
-        1   2 -0.780949  0.693147
-        2   3 -0.418711  1.098612
-        3   4 -0.269708  1.386294
-        4   5 -0.274002  1.609438
-        5   6 -0.500792  1.791759
-        6   7  1.649697  1.945910
-        7   8 -1.495604  2.079442
-        8   9  0.549296  2.197225
-        9  10 -0.758542  2.302585
+            A    B      ln_A
+         0  1 -1.0  0.000000
+         1  2  0.5  0.693147
```

Comment:
I agree with @jorisvandenbossche that not using `lambda` would be better. But I don't think that's possible, given that the column from the dataframe need to be specified. So, unless there is an easy way to avoid lambda, I'd prefer a function where the user can quickly compute the result mentally, and understand faster what happened in the operation. And if it's a real example, even better, I think that makes things even simpler for the reader.

Not sure if this is too complex, but something like that would be better in my opinion:

```
import pandas as pd

df = pd.DataFrame([('liquid', 100.),
                   ('liquid', 356.73),
                   ('gas', -252.87)],
                  index=['water', 'mercury', 'hydrogen'],
                  columns=('state', 'boiling_point_c'))

df.assign(state=lambda x: pd.Categorical(x.state),
          boiling_point_f=lambda x: x.boiling_point_c * 9 / 5 + 32)
```